pkg/util/steps/steps.go

```diff
@@ -0,0 +1,180 @@
+package steps
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"runtime"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/azureerrors"
+	"github.com/Azure/ARO-RP/pkg/util/refreshable"
+)
+
+// friendlyName returns a "friendly" stringified name of the given func.
+func friendlyName(f interface{}) string {
+	return runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name()
+}
+
```

Comment:
this file looks a bit of a hold-all. maybe have a file for each step? also remove the onlyinDev/Prod until you need them